docs/sdks/languages/java.mdx

```diff
@@ -88,13 +89,22 @@ public class App {
             An access token obtained from the machine identity login endpoint.
         </ParamField>
 
+        <ParamField query="setCacheTTL()" type="number" default="300" optional>
+            Time-to-live (in seconds) for refreshing cached secrets.
+            If manually set to 0, caching will be disabled, this is not recommended.
+        </ParamField>
+
         <ParamField query="setSiteURL()" type="string" default="https://app.infisical.com" optional>
             Your self-hosted absolute site URL including the protocol (e.g. `https://app.infisical.com`)
         </ParamField>
     </Expandable>
 
 </ParamField>
 
+### Caching
+
+The SDK caches every secret and updates it periodically based on the provided cacheTTL. For example, if cacheTTL of 300 is provided, then a secret will be refetched every 5 minutes. The SDK first attempts to return the cached secret, but if no secret is found it'll make a request to get the fresh secret. Every time the secret is fetched from the cache, the cached secret expiry is reset.
```

Comment:
Is there actually a periodic check? Or do you mean that a expiry is set on the cached secrets and when a request to fetch secrets is made and it exceeds a cached secret ttl, fresh secrets will be fetched?